pkg/controller/cloudlet_api.go

```diff
@@ -2857,3 +2857,66 @@ func redactAccessVars(accessVars map[string]string) map[string]string {
 	}
 	return vars
 }
+
+func canRunRefreshCerts(cloudlet *edgeproto.Cloudlet, features *edgeproto.PlatformFeatures) bool {
+	// For CRM on edge, CRM runs cert refresh
+	return !cloudlet.CrmOnEdge
+}
+
+func (s *CloudletApi) RefreshCerts(key *edgeproto.CloudletKey, cb edgeproto.CloudletApi_RefreshCertsServer) error {
+	ctx := cb.Context()
+	ptof := s.all.platformFeaturesApi.FeaturesByPlatform()
+
+	filter := &edgeproto.Cloudlet{
+		Key: *key,
+	}
+	cloudlets := []edgeproto.Cloudlet{}
+	err := s.cache.Show(filter, func(obj *edgeproto.Cloudlet) error {
```

Comment:
Can this be a partial key? If this is a complete cloudlet key, there is only a single cloudlet to consider and `s.Get()` would be a better fit.

Reply:
Yup, it's intended to allow for a partial key specified by the user.